src/main/scala/com/datamountaineer/streamreactor/connect/schemas/SinkRecordConverterHelper.scala

```diff
@@ -0,0 +1,186 @@
+      newStruct
+    }
+
+
+    /**
+     * make new sink record, taking fields
+     * from the key, value and headers
+     * */
+    def newFilteredRecord(
+        fields: Map[String, String],
+        ignoreFields: Set[String] = Set.empty[String],
+        keyFields: Map[String, String] = Map.empty[String, String],
+        headerFields: Map[String, String] = Map.empty[String, String],
+        retainKey: Boolean = false,
+        retainHeaders: Boolean = false): SinkRecord = {
+
+      //if we have keys fields and a key value extract
+      val keyStruct = if (keyFields.nonEmpty && record.key() != null) {
+        extract(payload = record.key(),
+                payloadSchema = record.keySchema(),
+                key = true,
+                fields = keyFields,
+                ignoreFields = Set.empty)
+      } else {
+        logger.warn(
+          s"Key is null for topic [${record.topic()}], partition [${record
+            .kafkaPartition()}], offset [${record.kafkaOffset()}])")
+        new Struct(SchemaBuilder.struct().build())
+      }
+
+      //if we have value fields and a value extract
+      val valueStruct = if (fields.nonEmpty && record.value() != null) {
+        extract(payload = record.value(),
+                payloadSchema = record.valueSchema(),
+                key = false,
+                fields = fields,
+                ignoreFields = ignoreFields)
+      } else {
+        logger.warn(
+          s"Value is null for topic [${record.topic()}], partition [${record
+            .kafkaPartition()}], offset [${record.kafkaOffset()}])")
+        new Struct(SchemaBuilder.struct().build())
+      }
+
+      //create a new struct with the keys, values and headers
+      val struct = keyStruct ++ valueStruct ++ structFromHeaders(headerFields)
+
+      new SinkRecord(
+        record.topic(),
+        record.kafkaPartition(),
+        if (retainKey) record.keySchema() else null,
+        if (retainKey) record.key() else null,
+        struct.schema(),
+        struct,
+        record.kafkaOffset(),
+        record.timestamp(),
+        record.timestampType(),
+        if (retainHeaders) record.headers() else new ConnectHeaders()
+      )
+    }
+
+    private def extract(payload: Object,
+                        payloadSchema: Schema,
+                        key: Boolean,
+                        fields: Map[String, String],
+                        ignoreFields: Set[String]): Struct = {
+
+      if (payloadSchema == null) {
+        //json with no schema
+        val j: util.Map[String, Any] = convertSchemalessJson(
+          record = record,
+          fields = fields,
+          ignoreFields = ignoreFields,
+          key = key,
+          includeAllFields = fields.contains("*")
+        )
+
+        val newSchema = SchemaBuilder.struct()
+        j.asScala
+          .foreach {
+            case (name, value) => newSchema.field(name, Schema.STRING_SCHEMA)

```

Comment:
This is not covering a scenario where:
 1. value is a primitive INT/LONG/etc
  2. value can be itself a util.Map. This would happen for nested jsons